torch/csrc/jit/codegen/cuda/runtime/tensorcore.cu

```diff
@@ -0,0 +1,180 @@
+        "=r"(_C[3]),
+        "=r"(_C[4]),
+        "=r"(_C[5]),
+        "=r"(_C[6]),
+        "=r"(_C[7])
+      : "r"(_A[0]),
+        "r"(_A[1]),
+        "r"(_B[0]),
+        "r"(_B[1]),
+        "r"(_C[0]),
+        "r"(_C[1]),
+        "r"(_C[2]),
+        "r"(_C[3]),
+        "r"(_C[4]),
+        "r"(_C[5]),
+        "r"(_C[6]),
+        "r"(_C[7]));
+}
+
+DEVICE_INLINE void mmaM8n8k4tn(
+    Array<float, 8>* C,
+    Array<__half, 4>* A,
+    Array<__half, 4>* B) {
+  unsigned const* _A = reinterpret_cast<unsigned const*>(A);
+  unsigned const* _B = reinterpret_cast<unsigned const*>(B);
+  unsigned* _C = reinterpret_cast<unsigned*>(C);
+
+  asm("mma.sync.aligned.m8n8k4.row.col.f32.f16.f16.f32 {%0,%1,%2,%3,%4,%5,%6,%7}, {%8,%9}, {%10,%11}, {%12,%13,%14,%15,%16,%17,%18,%19};\n"
+      : "=r"(_C[0]),
+        "=r"(_C[1]),
+        "=r"(_C[2]),
+        "=r"(_C[3]),
+        "=r"(_C[4]),
+        "=r"(_C[5]),
+        "=r"(_C[6]),
+        "=r"(_C[7])
+      : "r"(_A[0]),
+        "r"(_A[1]),
+        "r"(_B[0]),
+        "r"(_B[1]),
+        "r"(_C[0]),
+        "r"(_C[1]),
+        "r"(_C[2]),
+        "r"(_C[3]),
+        "r"(_C[4]),
+        "r"(_C[5]),
+        "r"(_C[6]),
+        "r"(_C[7]));
+}
+
+DEVICE_INLINE void mmaM8n8k4nt(
+    Array<float, 8>* C,
+    Array<__half, 4>* A,
+    Array<__half, 4>* B) {
+  unsigned const* _A = reinterpret_cast<unsigned const*>(A);
+  unsigned const* _B = reinterpret_cast<unsigned const*>(B);
+  unsigned* _C = reinterpret_cast<unsigned*>(C);
+
+  asm("mma.sync.aligned.m8n8k4.col.row.f32.f16.f16.f32 {%0,%1,%2,%3,%4,%5,%6,%7}, {%8,%9}, {%10,%11}, {%12,%13,%14,%15,%16,%17,%18,%19};\n"
+      : "=r"(_C[0]),
+        "=r"(_C[1]),
+        "=r"(_C[2]),
+        "=r"(_C[3]),
+        "=r"(_C[4]),
+        "=r"(_C[5]),
+        "=r"(_C[6]),
+        "=r"(_C[7])
+      : "r"(_A[0]),
+        "r"(_A[1]),
+        "r"(_B[0]),
+        "r"(_B[1]),
+        "r"(_C[0]),
+        "r"(_C[1]),
+        "r"(_C[2]),
+        "r"(_C[3]),
+        "r"(_C[4]),
+        "r"(_C[5]),
+        "r"(_C[6]),
+        "r"(_C[7]));
+}
+
+// TODO: in a follow up,
+//    lift this part onto iterdomain ops, once the
+//    swizzle ops are ready.
+template <int acc_stride>
+DEVICE_INLINE Array<float, 8> accToMma(float* _C) {
+  float C_data[8] = {
+      _C[0],
+      _C[1],
+      _C[acc_stride],
+      _C[acc_stride + 1],
+      _C[2],
+      _C[3],
+      _C[acc_stride + 2],
+      _C[acc_stride + 3],
+  };
+
+  return *reinterpret_cast<Array<float, 8>*>(&C_data[0]);
+}
+
+template <int acc_stride>
+DEVICE_INLINE void mmaToAcc(float* _C, Array<float, 8>& C) {
+  float* C_data = reinterpret_cast<float*>(&C);
+  _C[0] = C_data[0];
+  _C[1] = C_data[1];
+  _C[acc_stride] = C_data[2];
+  _C[acc_stride + 1] = C_data[3];
+  _C[2] = C_data[4];
+  _C[3] = C_data[5];
+  _C[acc_stride + 2] = C_data[6];
+  _C[acc_stride + 3] = C_data[7];
+}
+
+} // namespace util
+
+template <int acc_stride>
+DEVICE_INLINE void M16N16K4TT(
```

Comment:
@rdspring1 FYI, Volta mma's have options to use row/col major layouts for operands.